draft-ietf-dnsop-svcb-httpssvc.md

```diff
@@ -938,8 +938,15 @@ record, groups of clients will necessarily receive the same
 SvcFieldValue.  Therefore, HTTPSSVC is not suitable for uses that
 require single-client granularity.
 
+## Interaction with Alt-Svc
+
 If the client has an Alt-Svc cache, and a usable Alt-Svc value is
 present in that cache, then the client MAY skip the HTTPSSVC query.
+If Alt-Svc connection fails, the client SHOULD fall back to the HTTPSSVC
+client connection procedure ({{client-behavior}}).
```

Comment:
This text still allows clients to short-circuit HTTPSSVC if an Alt-Svc cache entry exists, which means all the issues from #105 apply. (Is this just a mistake? The client behavior below seems to suggest HTTPSSVC goes first.)

Or is the intent that the client behavior overrides this sentence? It's probably worth writing this a bit more explicitly. The combination is also weird because that text isn't additional requirements on top of the base algorithm. It completely changes it. (The base algorithm, while not explicitly spelled out, implies that the Alt-Svc integration point is before HTTPSSVC while the ESNI algorithm says it's after.)

Reply:
> Or is the intent that the ESNI client behavior overrides this paragraph? It's probably worth writing this a bit more explicitly.

Yes.  I've pushed a change to clarify this.

Reply:
It seems like we should be explicit around whether DNS lookups for the Alt-Svc host are done only for A/AAAA records or also for SVCB/HTTPSSVC records?
Approaches:
1) Be explicit that the host in Alt-Svc is just resolved for A/AAAA records/
2) Have an Alt-Svc attribute of "svcb=1" which would indicate that the Alt-Svc name is to a HTTPSSVC record?  In the behavior ordering, we could include then allow these to go before HTTPSSVC DNS lookups.  This might make sense in a separate AltSvc-bis/rfc7838bis draft that updates this and 7838 (along with other changes).

Reply:
I think your option 1 is our only choice at this point.  I've uploaded a sentence to clarify that.

Some kind of interaction between Alt-Svc and SVCB could be very interesting in the future, but as it stands, I don't see how to make that delegation work.  For example, there could be very complex interactions between Alt-Svc and HTTPSSVC values that can disagree (e.g. ALPN/"transport", "persist", "esniconfig").

I think Alt-Svc-bis is beginning to sound like a good idea, but first we should get this draft finalized.